src/libparms/db/parmsdb.cpp

```diff
@@ -918,352 +908,75 @@ bool ParmsDb::prepare() {
+    if (!prepareQuery(SELECT_DRIVE_BY_DRIVEID_REQUEST_ID, SELECT_DRIVE_BY_DRIVEID_REQUEST)) return false;
+    if (!prepareQuery(SELECT_ALL_DRIVES_REQUEST_ID, SELECT_ALL_DRIVES_REQUEST)) return false;
+    if (!prepareQuery(SELECT_ALL_DRIVES_BY_ACCOUNT_REQUEST_ID, SELECT_ALL_DRIVES_BY_ACCOUNT_REQUEST)) return false;
     // Sync
-    ASSERT(queryCreate(INSERT_SYNC_REQUEST_ID));
-    if (!queryPrepare(INSERT_SYNC_REQUEST_ID, INSERT_SYNC_REQUEST, false, errId, error)) {
-        queryFree(INSERT_SYNC_REQUEST_ID);
-        return sqlFail(INSERT_SYNC_REQUEST_ID, error);
-    }
-
-    ASSERT(queryCreate(UPDATE_SYNC_REQUEST_ID));
-    if (!queryPrepare(UPDATE_SYNC_REQUEST_ID, UPDATE_SYNC_REQUEST, false, errId, error)) {
-        queryFree(UPDATE_SYNC_REQUEST_ID);
-        return sqlFail(UPDATE_SYNC_REQUEST_ID, error);
-    }
-
-    ASSERT(queryCreate(UPDATE_SYNC_PAUSED_REQUEST_ID));
-    if (!queryPrepare(UPDATE_SYNC_PAUSED_REQUEST_ID, UPDATE_SYNC_PAUSED_REQUEST, false, errId, error)) {
-        queryFree(UPDATE_SYNC_PAUSED_REQUEST_ID);
-        return sqlFail(UPDATE_SYNC_PAUSED_REQUEST_ID, error);
-    }
-
-    ASSERT(queryCreate(UPDATE_SYNC_HASFULLYCOMPLETED_REQUEST_ID));
-    if (!queryPrepare(UPDATE_SYNC_HASFULLYCOMPLETED_REQUEST_ID, UPDATE_SYNC_HASFULLYCOMPLETED_REQUEST, false, errId, error)) {
-        queryFree(UPDATE_SYNC_HASFULLYCOMPLETED_REQUEST_ID);
-        return sqlFail(UPDATE_SYNC_HASFULLYCOMPLETED_REQUEST_ID, error);
-    }
-
-    ASSERT(queryCreate(DELETE_SYNC_REQUEST_ID));
-    if (!queryPrepare(DELETE_SYNC_REQUEST_ID, DELETE_SYNC_REQUEST, false, errId, error)) {
-        queryFree(DELETE_SYNC_REQUEST_ID);
-        return sqlFail(DELETE_SYNC_REQUEST_ID, error);
-    }
-
-    ASSERT(queryCreate(SELECT_SYNC_REQUEST_ID));
-    if (!queryPrepare(SELECT_SYNC_REQUEST_ID, SELECT_SYNC_REQUEST, false, errId, error)) {
-        queryFree(SELECT_SYNC_REQUEST_ID);
-        return sqlFail(SELECT_SYNC_REQUEST_ID, error);
-    }
-
-    ASSERT(queryCreate(SELECT_SYNC_BY_PATH_REQUEST_ID));
-    if (!queryPrepare(SELECT_SYNC_BY_PATH_REQUEST_ID, SELECT_SYNC_BY_PATH_REQUEST, false, errId, error)) {
-        queryFree(SELECT_SYNC_BY_PATH_REQUEST_ID);
-        return sqlFail(SELECT_SYNC_BY_PATH_REQUEST_ID, error);
-    }
-
-    ASSERT(queryCreate(SELECT_ALL_SYNCS_REQUEST_ID));
-    if (!queryPrepare(SELECT_ALL_SYNCS_REQUEST_ID, SELECT_ALL_SYNCS_REQUEST, false, errId, error)) {
-        queryFree(SELECT_ALL_SYNCS_REQUEST_ID);
-        return sqlFail(SELECT_ALL_SYNCS_REQUEST_ID, error);
-    }
-
-    ASSERT(queryCreate(SELECT_ALL_SYNCS_BY_DRIVE_REQUEST_ID));
-    if (!queryPrepare(SELECT_ALL_SYNCS_BY_DRIVE_REQUEST_ID, SELECT_ALL_SYNCS_BY_DRIVE_REQUEST, false, errId, error)) {
-        queryFree(SELECT_ALL_SYNCS_BY_DRIVE_REQUEST_ID);
-        return sqlFail(SELECT_ALL_SYNCS_BY_DRIVE_REQUEST_ID, error);
-    }
-
+    if (!prepareQuery(INSERT_SYNC_REQUEST_ID, INSERT_SYNC_REQUEST)) return false;
```

Comment:
Defining member variables instead of macros would enable us to loop over all requests of a special type, for instance. Maybe you didn't go that way (i.e., defining static member variables) because it would bring you much too far in the refactoring direction? Or is there a reason not to do so?

Reply:
Yes, that PR is becoming quite big while it should just have been adding a simple method to check wheather a column exist or not.
Let's push further refactoring in a dedicated task if it is ok for you: https://infomaniak.atlassian.net/browse/KDESKTOP-1296